python/cuml/fil/fil.pyx

```diff
@@ -0,0 +1,232 @@
+                   forest_t)
+
+    cdef void predict(cumlHandle& handle,
+                      forest_t,
+                      float*,
+                      float*,
+                      size_t)
+
+    cdef forest_t from_treelite(cumlHandle& handle,
+                                forest_t*,
+                                ModelHandle,
+                                treelite_params_t*)
+
+cdef class FIL_impl():
+
+    cpdef object handle
+    cdef forest_t forest_data
+    cdef object algo
+    cdef object threshold
+
+    def __cinit__(self,
+                  algo=0,
+                  threshold=0.0,
+                  handle=None):
+        self.algo = algo
+        self.threshold = threshold
+        self.handle = handle
+
+    def predict(self, X, preds=None):
+        cdef uintptr_t X_ptr
+        X_m, X_ptr, n_rows, _, X_dtype = \
+            input_to_dev_array(X, order='C')
+
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+
+        if preds is None:
+            preds = cudf.Series(zeros(n_rows, dtype=np.float32))
+        cdef uintptr_t preds_ptr
+        preds_m, preds_ptr, _, _, _ = \
+            input_to_dev_array(preds)
+        predict(handle_[0],
+                self.forest_data,
+                <float*> preds_ptr,
+                <float*> X_ptr,
+                <size_t> n_rows)
+        self.handle.sync()
+        # synchronous w/o a stream
+        return preds
+
+    def from_treelite(self, model, output_class):
+        cdef treelite_params_t treelite_params
+        treelite_params.output_class = output_class
+        treelite_params.threshold = self.threshold
+        treelite_params.algo = self.algo
+        cdef forest_t* forest_pointer =\
+            <forest_t*><size_t> model.handle.value
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        cdef uintptr_t model_ptr = model.handle.value
+        self.forest_data = from_treelite(handle_[0],
+                                         forest_pointer,
+                                         <ModelHandle> model_ptr,
+                                         &treelite_params)
+        return self
+
+    def __cdel__(self):
+        cdef cumlHandle* handle_ =\
+            <cumlHandle*><size_t>self.handle.getHandle()
+        free(handle_[0],
+             self.forest_data)
+        return self
+
+
+class FIL(Base):
+    """
+    Parameters
+    ----------
+    model : the model information in the treelite format
+       loaded from a saved model using the treelite API
+       https://treelite.readthedocs.io/en/latest/treelite-api.html
+    output_class: boolean
+       True or False
+    algo : 0 = NAIVE, 1 = TREE_REORG, 2 = BATCH_TREE_REORG
+    threshold : threshold is used to for classification
+       if output == OUTPUT_CLASS, else it is ignored
+    """
+    def __init__(self,
+                 algo=0,
+                 threshold=0.0,
```

Comment:
output or output_class is mentioned in the docs but not implemented here?

Reply:
`algo`, `threshold` and `output_class` belong to the list of parameters of `from_treelite()`.